### PR TITLE
Add E2E test for content types

### PR DESCRIPTION
**Pull request by shinnlok**
Few notes:

- The second test case is disabled because the feature doesn't appear to be functional
- I couldn't think of a better way to make the virtual table load everything in the list with Cypress
- Not sure where/if I should be adding `data-test` attributes

### CODE REVIEW

File: apps/sensenet/cypress/integration/content-types/content-types.spec.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { pathWithQueryParams } from '../../../src/services/query-string-builder'
+
+describe('Content types', () => {
+  beforeEach(() => {
+    cy.login()
+    cy.visit(pathWithQueryParams({ path: '/', newParams: { repoUrl: Cypress.env('repoUrl') } }))
+      .get('div[aria-label="Content Types"]')
+      .click()
+    cy.get('.ReactVirtualized__Grid__innerScrollContainer')
```
Comment (herflis) [inline apps/sensenet/cypress/integration/content-types/content-types.spec.ts:9]: This row seems to be not used later so you can remove it
```suggestion
```

File: apps/sensenet/cypress/integration/content-types/content-types.spec.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { pathWithQueryParams } from '../../../src/services/query-string-builder'
+
+describe('Content types', () => {
+  beforeEach(() => {
+    cy.login()
+    cy.visit(pathWithQueryParams({ path: '/', newParams: { repoUrl: Cypress.env('repoUrl') } }))
+      .get('div[aria-label="Content Types"]')
```
Comment (herflis) [inline apps/sensenet/cypress/integration/content-types/content-types.spec.ts:7]: This selector should be changed to a `data-test=Content Types`. This attribute is already added to the drawer component by another contributor so you have to change only the selector here.

File: apps/sensenet/cypress/integration/content-types/content-types.spec.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      })
+
+    cy.get('.MuiPopover-root').click()
+  })
+
+  it('double clicking on article should open binary editor', () => {
+    cy.get('@articleRow').dblclick()
+    cy.get('div').contains('Article').should('be.visible')
+    cy.get('.monaco-editor').should('be.visible')
+    cy.get('button[aria-label="Cancel"]').click()
```
Comment (herflis) [inline apps/sensenet/cypress/integration/content-types/content-types.spec.ts:38]: Put data-test="cancel" to the cancel button in edit-view.tsx. It is in a separate package in this repository (sn-client), you can find it in /packages/sn-controls-react/viewcontrols folder.
And then of course please change here the selector in the test for data-test

**Review by herflis — CHANGES_REQUESTED** [review]
Yes you're right the second case cannot reproduce now, so I will remove it from the issue, thanks for pointing this out,
I've added some suggestions to your code, one is an unused row, others are related to the data-test attribute that you've mentioned. After changing the selector you will not be able to run these tests successfully on your test repository, but don't worry about it because otherwise your pr is ok 👍

File: apps/sensenet/cypress/integration/content-types/content-types.spec.ts (hunk 2 above)
Comment (shinnlok) [reply apps/sensenet/cypress/integration/content-types/content-types.spec.ts]: Done

**Comment by herflis** [comment]
@shinnlok thank you for taking the time to help us! We greatly appreciate your contribution, you will get the extra sensenet-Hacktoberfest swag 🧦. We will contact you via email after the Hacktoberfest event. Hope you are having a great Hacktoberfest! 🎉 🙂

**Pull request merged**